src/chia_dialect.rs

```diff
@@ -103,4 +107,12 @@ impl Dialect for ChiaDialect {
     fn apply_kw(&self) -> &[u8] {
         &[2]
     }
+
+    fn val_stack_limit(&self) -> usize {
+        if (self.flags & LIMIT_HEAP) != 0 {
+            10000000
```

Comment:
something that might be useful here is to take some big singleton inner puzzle and do treehash on it under this limit.  i'm thinking back to the recent performance improvement in clvm that might change the order of retirements from the stack.  i don't definitely know there'll be a lot of values left on the stack before the final sha256, but it'd be good to have a smoke test that it generally isn't blowing up with the limit on.

Reply:
I tested the whole chain with this limit, so we don't have any puzzles this large that have run on the chain so far at least